src/arch/ArchHooks/ArchHooks_Win32Static.cpp

```diff
@@ -30,22 +29,19 @@ static void InitTimer()
 		return;
 	g_bTimerInitialized = true;
 
-	// Grab important information for QPC during the timer initialization
+	/* Grab important information for QPC during
+	 * the timer initialization. The frequency is
+	 * constant, so we only need to grab it once. */
 	QueryPerformanceCounter(&g_liStartTime);
 	QueryPerformanceFrequency(&g_liFrequency);
 }
 
-std::int64_t ArchHooks::GetMicrosecondsSinceStart(bool bAccurate)
+std::uint64_t ArchHooks::GetMicrosecondsSinceStart(bool bAccurate)
 {
-	// Make sure the timer is initialized
 	if (!g_bTimerInitialized)
 		InitTimer();
-
-	// Get the current time
-	QueryPerformanceCounter(&g_liCurrentTime);
-
-	// Calculate the elapsed time in microseconds.
-	return ((g_liCurrentTime.QuadPart - g_liStartTime.QuadPart) * 1000000.0) / g_liFrequency.QuadPart;
+    QueryPerformanceCounter(&g_liCurrentTime);
+    return (g_liCurrentTime.QuadPart - g_liStartTime.QuadPart) * 1000000ULL / g_liFrequency.QuadPart;
```

Comment:
I think you lose precision by only working with integers (which is why we multiply by 1000000.0) before returning an int64 (which will autocast to that type)

Reply:
I previously thought we were going to benefit from floating point here, but now I'm not so sure.
I looked into this some more before making this change & this is how I came to this decision.

1. high profile projects (Intel, Android, libusb) are using 1000000ULL instead of 1000000.0 when using QPC and storing or returning an integer value of microseconds
2. even when this was returning int64_t, we were still losing the fractional part before it got to RageTimer
3. The Microsoft documentation suggests to do integer math instead of floating point math here when the values are stored in integer data types.

https://learn.microsoft.com/en-us/windows/win32/sysinfo/acquiring-high-resolution-time-stamps#virtualization
> LARGE_INTEGER StartingTime, EndingTime, ElapsedMicroseconds;
> LARGE_INTEGER Frequency;
...
> ElapsedMicroseconds.QuadPart *= 1000000;
> ElapsedMicroseconds.QuadPart /= Frequency.QuadPart;

I think it makes sense to use integer math here because we are returning an integer, the FPS boost from doing so seems like a nice benefit, and calculating the time as integer math is sufficiently precise for our needs.

I would argue it is better for our needs to have this function return faster with a suitably precise answer, rather than taking longer to return the most accurate possible answer.